traffic_monitor/cache/astats.go

```diff
@@ -116,8 +119,22 @@ func astatsParse(cacheName string, rdr io.Reader, pollCTX interface{}) (Statisti
 		astats.Ats["system.proc.loadavg"] = astats.System.ProcLoadavg
 		astats.Ats["system.proc.net.dev"] = astats.System.ProcNetDev
 
+		via := ctx.HTTPHeader.Get(rfc.Via)
+		if via != "" {
+			viaRegexSubmatch := regexp.MustCompile(` ([^.]+)`).FindStringSubmatch(via)
+			if len(viaRegexSubmatch) > 0 {
+				astats.Ats[rfc.Via] = viaRegexSubmatch[1]
+			}
+		}
 		return stats, astats.Ats, nil
 	} else if ctype == "text/csv" {
+		via := ctx.HTTPHeader.Get(rfc.Via)
+		if via != "" {
+			viaRegexSubmatch := regexp.MustCompile(` ([^.]+)`).FindStringSubmatch(via)
```

Comment:
`regexp.MustCompile()` compiles the regex. The compiled regex should be reused, rather than calling `regexp.MustCompile()` each time.